packages/cli/src/ui/components/InputPrompt.tsx

```diff
@@ -402,12 +466,14 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
         buffer.moveToOffset(cpLen(buffer.text));
         return;
       }
-      // Ctrl+C (Clear input)
+      // Ctrl+C (Clear input or trigger exit)
       if (key.ctrl && key.name === 'c') {
         if (buffer.text.length > 0) {
           buffer.setText('');
           resetCompletionState();
-          return;
+        } else if (onCtrlCWithEmptyBuffer) {
+          // Buffer is empty, notify parent component to handle exit
+          onCtrlCWithEmptyBuffer();
```

Comment:
why this change? we shouldn't be too eager to shut down the app. If we want this lets put it in a separate bug plus pr to discuss.